posts/2021-08-30-sharing-data-with-fine-grained-control.md

```diff
@@ -0,0 +1,792 @@
+               },
+               {
+                   name: "Granulated Sugar",
+                   quantity: "833 grams"
+               }
+           ],
+           directions: [
+               "Mix dry ingredients",
+               "Bake",
+               "Let cupcakes cool for 20min",
+               "Make icing",
+               "Put icing on cupcakes",
+               "Profit"
+           ]
+       },
+       aclInput: {
+           acl: [
+               {
+                   principal: {
+                       nodes: [ "*" ]
+                   },
+                   path: "name",
+                   operations: [ READ ]
+               }
+           ]
+       }
+   ) {
+       error
+       result {
+           id
+           node_owner
+       }  
+   }
+}
+```
+</details>
+
+In this example, we use **"*"** for our nodes. This is a short-hand representation of all current and future nodes in our Uni. Alternately, we could have specified each node as an item in a list like so:
+
+```json
+principal: {
+    nodes: [ "Bob", "Eve" ]
+}
+```
+
+When `Alice` inserts this new recipe, `Bob` and `Eve` will be able to view some, but not all, properties. `Bob` and `Eve` will be able to read the **name** of the recipe. However, they will not be able to see any of the other attributes since `Alice` didn't give either of them access to those properties. They will also not be able to update or delete the recipe since they both only have **READ** access.  Let's run the following query from the GraphQL Explorer associated with either `Bob` or `Eve` to confirm the behavior.
+
+<details>
+<summary>listRecipes Query</summary>
+
+Run this query from either the `Bob` or `Eve` GraphQL Explorer.
+
+```graphql
+query listRecipes {
+  listRecipes {
+    Recipes {
+      ... on Recipe {
+        id
+        name
+        price
+        sku
+        recipeType
+        recipeYield
+        directions
+        ingredients {
+          name
+          quantity
+        }
+      }
+      ... on Recipe_Partial {
+        id
+        name
+        price
+        sku
+        recipeType
+        recipeYield
+        directions
+        ingredients {
+          name
+          quantity
+        }
+      }
+    }
+  }
+}
+```
+</details>
+
+We will receive results that look similar to the following:
+
+```json
+​​{
+  "data": {
+    "listRecipes": {
+      "Recipes": [
+        {
+          "id": "017b70eb-6d92-7039-4507-cc48eddb3c7c",
+          "name": "Sprinkles Cupcake",
+          "price": null,
+          "sku": null,
+          "recipeType": null,
+          "recipeYield": null,
+          "directions": null,
+          "ingredients": null
+        }
+      ]
+    }
+  }
+}
+```
+
+Notice that only the **name** is visible; the rest of the fields are displayed as null.
+
+![Sharing Selected Properties](https://d24nhiikxn5jns.cloudfront.net/images/blogs/2021-08-30-sharing-data-with-fine-grained-control/01-fine-grained-share-selected-properties.png)
```

Comment:
Just for my own info...in which AWS account is this CloudFront distro (and S3 backing store) based?

Reply:
I'll share that information.